config/boot.rb

```diff
@@ -1,3 +1,14 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
-require 'bundler/setup' # Set up gems listed in the Gemfile.
+require 'bundler/setup' if File.exist?(ENV['BUNDLE_GEMFILE'])
+
+require 'rails/commands/server'
+
+module Rails
+  class Server
+    alias :default_options_alias :default_options
+    def default_options
+      default_options_alias.merge!(:Host => '0.0.0.0')
```

Comment:
Can you create and env variable instead? Something like:

```rb
default_options_alias.merge!(Host: ENV.fetch('BIND_ADDR', 'localhost'))
```

Then set `BIND_ADDR` in vagrant somewhere. I don't want to default to `0.0.0.0` because of security reasons.